### PR TITLE
gtkwave: support darwin build

### PR DESCRIPTION
**Pull request by jiegec**
###### Description of changes

This pr addes darwin support to gtkwave. It requires fixes from https://github.com/gtkwave/gtkwave/pull/136 for building.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/gtkwave/default.nix [hunk 1]
```diff
@@ -23,7 +24,13 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkg-config wrapGAppsHook ];
-  buildInputs = [ bzip2 glib gperf gtk3 judy tcl tk xz ];
+  buildInputs = [ bzip2 glib gperf gtk3 judy tcl tk xz ]
+    ++ lib.optional stdenv.isDarwin gtk-mac-integration;
+
+  patches = [ ] ++ lib.optional stdenv.isDarwin [
+    ./0001-Fix-detection-of-quartz-in-gdk-3.0-target.patch
```
Comment (Mindavi) [inline pkgs/applications/science/electronics/gtkwave/default.nix:31]: Please use fetchpatch, and make the patches unconditional.

**Review by Mindavi — CHANGES_REQUESTED** [review]
Requested some changes. Also, let's give upstream some time to respond. You can ping me if there's any updates upstream (or if nobody says anything within say a week or 2 we can always just merge it here for now)

File: pkgs/applications/science/electronics/gtkwave/default.nix [hunk 2]
```diff
@@ -37,6 +44,6 @@ stdenv.mkDerivation rec {
     homepage = "http://gtkwave.sourceforge.net";
     license = lib.licenses.gpl2Plus;
     maintainers = with lib.maintainers; [ thoughtpolice ];
```
[the remark this answers is not in the document]
Comment (jiegec) [reply pkgs/applications/science/electronics/gtkwave/default.nix:46]: Sure, I will.

File: pkgs/applications/science/electronics/gtkwave/default.nix (hunk 1 above)
Comment (jiegec) [reply pkgs/applications/science/electronics/gtkwave/default.nix:31]: The pr commit applies to the master branch, but it cannot apply to the release tarball.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 178552` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gtkwave</li>
  </ul>
</details>

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 178552` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by jiegec** [comment]
Result of `nixpkgs-review pr 178552` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gtkwave</li>
  </ul>
</details>

**Comment by jiegec** [comment]
> Requested some changes. Also, let's give upstream some time to respond. You can ping me if there's any updates upstream (or if nobody says anything within say a week or 2 we can always just merge it here for now)

The upstream pr is merged.

**Review by veprbl — APPROVED** [review]
Diff LGTM

**Comment by Mindavi** [comment]
Not nice that we have to carry the patches but nothing to do about it

**Pull request merged**